### ISSUE TITLE
Cannot use peer authentication via connection string

### ISSUE THREAD

**Issue by jasonaowen** [joeandaverde/tinypg#25]
I can't seem to get TinyPg to connect with peer authentication. I believe this is a bug in TinyPg, and not in the underlying dependencies.

Peer auth connection strings often do not include a user name, but TinyPg seems to be explicitly defaulting to the `postgres` user:

https://github.com/joeandaverde/tinypg/blob/5e398c0152c12396957daf0b5c74b5115a4e0a49/src/tiny.ts#L25

More importantly, `Url.parse` doesn't know how to interpret either of the ways of representing socket connections that pg accepts: `postgresql://%2Fvar%2Frun%2Fpostgresql/db_name` or `postgresql:///db_name?host=/run/postgresql`; in both cases it reports that the URL contains no host, which triggers the TinyPg default of `localhost`:

https://github.com/joeandaverde/tinypg/blob/5e398c0152c12396957daf0b5c74b5115a4e0a49/src/tiny.ts#L32
https://github.com/joeandaverde/tinypg/blob/5e398c0152c12396957daf0b5c74b5115a4e0a49/src/tiny.ts#L38

(Eventually, pg may eventually accept the libpq standard of `postgresql:///db_name`, which would also be prevented by the current version of TinyPg; see also the conversation in https://github.com/brianc/node-postgres/issues/613 and related issues.)

In general, I think providing defaults for database connections is a somewhat perilous idea; I would rather know during development that I have not entered enough information. In this specific case, I don't think TinyPg needs to do any parsing of the connection string, as pg is able to correctly parse a `connectionString` passed in to `Pg.Pool`; I tested this by making this change in my `node_modules` directory, after which I was able to use TinyPg with both styles of connection string:

```diff
       const pool_options = _.isNil(options.pool_options) ? {} : options.pool_options

-      const config_from_url = parseConnectionConfigFromUrlOrDefault(options.connection_string, options.tls_options);
+      const config_from_url = { connectionString: options.connection_string };

       const pool_config: Pg.PoolConfig & { log: any } = {
```

If you like, I would be happy to clean this change up and put it into a pull request! I recognize that this would be a breaking change, and so would need a major version number bump.

**Comment by jasonaowen** [comment]
I should add that exporting `PGHOST=/run/postgresql` and `PGUSER=$(whoami)` does allow me to connect via peer authentication; the problem then is that I cannot do so purely with the connection string.

**Comment by joeandaverde** [comment]
Thank you for reporting this. I agree that the connection string should be fully compatible with the underlying driver as there's no benefit to restricting it. I also agree that defaults are a bad idea. I will try to update to reflect these ideas and major version bump soon.

**Comment by jasonaowen** [comment]
Fixed in #29. Thanks, @joeandaverde!
